### PR TITLE
fix: call reloadFeatureFlags on featureFlags

### PR DESCRIPTION
**Pull request by weyert**
## Changes

Call the `reloadFeatureFlags` on the `featureFlags`-object instead of `persistence`

## Checklist
- [ ] Tests for new code (if applicable)
- [ ] TypeScript definitions (module.d.ts) updated and in sync with library exports (if applicable)

### CODE REVIEW

File: src/posthog-core.js [hunk 1]
```diff
@@ -916,7 +916,7 @@ PostHogLib.prototype.reset = function (reset_device_id) {
     )
 
     // trigger onFeatureFlags callback
-    this.persistence.receivedFeatureFlags()
+    this.featureFlags.reloadFeatureFlags()
```
Comment (weyert) [inline src/posthog-core.js:919]: Maybe even just `this.reloadFeatureFlags()`? Not sure why it wasn't called before

**Comment by mariusandra** [comment]
Other than fixing a bug, this also introduces a new behaviour: we now make a request to `/decide` every time `.reset()` is called. This is probably what we want (new flags/variants for the new user, etc), but then making this request should be documented in a test. Is this something you can add?

**Comment by weyert** [comment]
> Other than fixing a bug, this also introduces a new behaviour: we now make a request to `/decide` every time `.reset()` is called. This is probably what we want (new flags/variants for the new user, etc), but then making this request should be documented in a test. Is this something you can add?

Ah yeah, I could also change it `receivedFeatureFlags()` like it was before but on the `featureFlags` instead of `persistence`? I have to admit I am not sure what the intention was of the original change

**Comment by weyert** [comment]
PR got reverted. Closing this one

**Pull request closed without merging**